### PR TITLE
chore: optimization debian package manager tweaks

### PR DESCRIPTION
**Pull request by rajpratik71**
By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages .

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

### CODE REVIEW

**Comment by Rajpratik71** [comment]
@blowmage @lukesneeringer @dazuma @jskeet  requesting review

**Comment by jskeet** [comment]
This isn't a repository I have any experience with. I defer to those who use it.

**Comment by Rajpratik71** [comment]
@busunkim96 requesting review

**Review by busunkim96 — APPROVED** [review]
@dazuma @chingor13 @bcoe @codyoss 

Could you confirm that the changes are OK for your language?

(If the images here aren't being used, please let me know and we can remove them to avoid confusion.)

**Comment by codyoss** [comment]
Go does not currently use this repository to manage its docker images used for builds(although maybe it should in the future). I defer to others.

**Review by dazuma — CHANGES_REQUESTED** [review]
Ruby's usage here is complicated, because there are two sources of these images (one here and the other in googleapis/google-cloud-ruby) and they compete with each other and can clobber each other. We need to clean that up, but in the meantime, please do not make any changes to any of the Ruby images in this repo, to prevent this repo from building Ruby images. If you want to make this change to the Ruby images, do so in https://github.com/googleapis/google-cloud-ruby/tree/master/.kokoro/docker instead.

**Comment by Rajpratik71** [comment]
> Ruby's usage here is complicated, because there are two sources of these images (one here and the other in googleapis/google-cloud-ruby) and they compete with each other and can clobber each other. We need to clean that up, but in the meantime, please do not make any changes to any of the Ruby images in this repo, to prevent this repo from building Ruby images. If you want to make this change to the Ruby images, do so in https://github.com/googleapis/google-cloud-ruby/tree/master/.kokoro/docker instead.

done as requested

**Review by dazuma — APPROVED** [review]

**Comment by dazuma** [comment]
Thanks!

**Review by bcoe — APPROVED** [review]

File: node/8-puppeteer/Dockerfile [hunk 1]
```diff
@@ -4,16 +4,16 @@ FROM node:8
...
 
 # See https://crbug.com/795759
-RUN apt-get update && apt-get install -yq libgconf-2-4
+RUN apt-get update && apt-get install -y --no-install-recommendsq libgconf-2-4
 
 # Install latest chrome dev package and fonts to support major charsets (Chinese, Japanese, Arabic, Hebrew, Thai and a few others)
 # Note: this installs the necessary libs to make the bundled version of Chromium that Puppeteer
 # installs, work.
-RUN apt-get update && apt-get install -y wget --no-install-recommends \
+RUN apt-get update && apt-get install -y --no-install-recommends wget --no-install-recommends \
```
Comment (dazuma) [inline node/8-puppeteer/Dockerfile:12]: Extra `--no-install-recommends`

File: node/8-puppeteer/Dockerfile [hunk 2]
```diff
@@ -4,16 +4,16 @@ FROM node:8
 # https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#running-puppeteer-in-docker
 
 # See https://crbug.com/795759
-RUN apt-get update && apt-get install -yq libgconf-2-4
+RUN apt-get update && apt-get install -y --no-install-recommendsq libgconf-2-4
```
Comment (dazuma) [inline node/8-puppeteer/Dockerfile:7]: Bad copy-paste

File: node/12-puppeteer/Dockerfile [hunk 3]
```diff
@@ -4,16 +4,16 @@ FROM node:12
...
 
 # See https://crbug.com/795759
-RUN apt-get update && apt-get install -yq libgconf-2-4
+RUN apt-get update && apt-get install -y --no-install-recommendsq libgconf-2-4
 
 # Install latest chrome dev package and fonts to support major charsets (Chinese, Japanese, Arabic, Hebrew, Thai and a few others)
 # Note: this installs the necessary libs to make the bundled version of Chromium that Puppeteer
 # installs, work.
-RUN apt-get update && apt-get install -y wget --no-install-recommends \
+RUN apt-get update && apt-get install -y --no-install-recommends wget --no-install-recommends \
```
Comment (dazuma) [inline node/12-puppeteer/Dockerfile:12]: Extra `--no-install-recommends`

File: node/12-puppeteer/Dockerfile [hunk 4]
```diff
@@ -4,16 +4,16 @@ FROM node:12
 # https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#running-puppeteer-in-docker
 
 # See https://crbug.com/795759
-RUN apt-get update && apt-get install -yq libgconf-2-4
+RUN apt-get update && apt-get install -y --no-install-recommendsq libgconf-2-4
```
Comment (dazuma) [inline node/12-puppeteer/Dockerfile:7]: Bad copy-paste

**Review by dazuma — DISMISSED** [review]

File: node/8-puppeteer/Dockerfile (hunk 1 above)
Comment (Rajpratik71) [reply node/8-puppeteer/Dockerfile]: fixed

File: node/8-puppeteer/Dockerfile (hunk 2 above)
Comment (Rajpratik71) [reply node/8-puppeteer/Dockerfile]: fixed

File: node/12-puppeteer/Dockerfile (hunk 3 above)
Comment (Rajpratik71) [reply node/12-puppeteer/Dockerfile]: fixed

File: node/12-puppeteer/Dockerfile (hunk 4 above)
Comment (Rajpratik71) [reply node/12-puppeteer/Dockerfile]: fixed

**Comment by JustinBeckwith** [comment]
@chingor13 - can I trouble you to give this one a review and decide if it's a change we want?

**Review by chingor13 — APPROVED** [review]
This looks fine to me.

**Comment by parthea** [comment]
Hi @Rajpratik71 , Please could you resolve conflicts ?

**Review by parthea — APPROVED** [review]

**Comment by parthea** [comment]
@Rajpratik71 The `conventionalcommits.org` check is failing. Please can you take a look? Changing the commit message to `chore: optimization debian package manager tweaks` may resolve the issue.

**Comment by Rajpratik71** [comment]
> @Rajpratik71 The `conventionalcommits.org` check is failing. Please can you take a look? Changing the commit message to `chore: optimization debian package manager tweaks` may resolve the issue.

@parthea  done as required

**Review by bcoe — APPROVED** [review]

**Comment by tmatsuo** [comment]
So are we pushing docker images for testing without knowing our tests would pass or not?
I would say this again and again. This is a recipe for disaster :(

**Comment by parthea** [comment]
I'm going to close this PR as there is potential risk that downstream tests will fail and the risk is not worth the benefit of smaller downloads.

**Pull request closed without merging**